src/otoole/read_strategies.py

```diff
@@ -174,6 +179,15 @@ def read(
                 input_data, config_type=config_type
             )
 
+        if self.write_defaults:
+            names = [
+                x for x in self.user_config if self.user_config[x]["type"] == "param"
+            ]
+            for name in names:
+                input_data[name] = self._expand_dataframe(
+                    name, input_data, default_values
+                )
+
```

Comment:
Here is where results are being excluded from default expansion. This is implemented for each read strategy. I implemented this just for testing, and needs to be expanded to handle result processing